### ISSUE TITLE
Slurm constraint not working

### ISSUE THREAD

**Issue by weber8thomas** [snakemake/snakemake#2631]
**Snakemake version**

8.2.0

**Describe the bug**

I need to specify a node CPU architecture for some of my rules and I cannot use the constraint parameter using slurm.  

**Logs**

Profile:

```
executor: slurm
# slurm: True

# Default ressources for all rules
default-resources:
  - slurm_account="korbel"
  - slurm_partition="htc"
  - qos="normal"
  - mem_mb=1000
  - runtime=60
  - constraint=""

set-resources:
  ashleys_mark_duplicates:
    constraint: "rome"

max-jobs-per-second: 100
max-status-checks-per-second: 100
local-cores: 1
latency-wait: 60
jobs: 200

# 5 attempts to restart to correspond to the progressive memory increase solving OOM
# restart-times: 5
keep-going: True
rerun-incomplete: True

software-deployment-method: conda
conda-frontend: "conda"

```

Snakemake log:

```
[Wed Jan 17 15:50:13 2024]
rule ashleys_mark_duplicates:
    input: .tests/data_CHR17/RPE-BM510/bam/BM510x04_PE20313.bam.sort
    output: .tests/data_CHR17/RPE-BM510/bam/BM510x04_PE20313.sort.mdup.bam
    log: .tests/data_CHR17/RPE-BM510/log/markdup/BM510x04_PE20313.log
    jobid: 0
    reason: Forced execution
    wildcards: folder=.tests/data_CHR17, sample=RPE-BM510, cell=BM510x04_PE20313
    resources: mem_mb=16000, mem_mib=15259, disk_mb=1000, disk_mib=954, tmpdir=<TBD>, slurm_account=korbel, slurm_partition=htc, qos=normal, runtime=600, constraint=rome
```

Job log:

```
Config file config/config.yaml is extended by additional config specified via the command line.
localrules directive specifies rules that are not present in the Snakefile:
	symlink_bam_ashleys
	ashleys_genecore_symlink
	ashleys_symlink_selected_bam

Building DAG of jobs...
Your conda installation is not configured to use strict channel priorities. This is however crucial for having robust and correct environments (for details, see https://conda-forge.org/docs/user/tipsandtricks.html). Please consider to configure strict priorities by executing 'conda config --set channel_priority strict'.
Using shell: /usr/bin/bash
Provided cores: 2
Rules claiming more threads will be scaled down.
Provided resources: mem_mb=16000, mem_mib=15259, disk_mb=1000, disk_mib=954
Singularity containers: ignored
Select jobs to execute...
InputFunctionException in rule ashleys_mark_duplicates in file /g/korbel2/weber/workspace/ashleys-qc-pipeline/workflow/rules/rules.smk, line 129:
Error:
  WorkflowError:
    Failed to evaluate default resources value '<function eval_resource_expression.<locals>.callable at 0x7ff26c90fe20>'.
        String arguments may need additional quoting. Ex: --default-resources "tmpdir='/home/user/tmp'".
    SyntaxError: invalid syntax (<string>, line 1)
Wildcards:
  folder=.tests/data_CHR17
  sample=RPE-BM510
  cell=BM510x04_PE20316
Traceback:
 (rule ashleys_mark_duplicates, line 230, /g/korbel2/weber/workspace/ashleys-qc-pipeline/workflow/rules/rules.smk)
```

**Minimal example**

**Additional context**

Was working without slurm mode (just a profile with the cluster definition) in snakemake 7.32.4

**Comment by Zaffe24** [comment]
I experienced the same problem with snakemake 7.32.4.
Including `slurm: True` in the profile caused `set-resources` parameters to be ignored in favor of  `default-parameters`.

**Comment by johanneskoester** [comment]
This is fixed in 8.10, but a (hopefully) final second fix comes in PR #2778.
